src/test/java/org/folio/rest/api/RequestsApiTest.java

```diff
@@ -1500,6 +1505,66 @@ public void canDeleteARequest()
     checkNotFound(requestStorageUrl(String.format("/%s", id)));
   }
 
+  @Test
+  public void awaitingPickupRequestClosedDateIsPresentAfterStatusUpdateFromOpenAwaitingPickupToClosedPickupExpired()
+    throws MalformedURLException, InterruptedException, ExecutionException, TimeoutException {
+
+    JsonObject request = new RequestRequestBuilder()
+      .withStatus(OPEN_AWAITING_PICKUP)
+      .create();
+
+    String requestId = createEntity(request, requestStorageUrl()).getJson().getString("id");
+    request.put("status", CLOSED_PICKUP_EXPIRED);
+
+    CompletableFuture<Response> putCompleted = new CompletableFuture<>();
+    client.put(requestStorageUrl("/" + requestId), request, TENANT_ID, ResponseHandler.empty(putCompleted));
+    putCompleted.get(5, TimeUnit.SECONDS);
+
+    JsonObject updatedRequest = getById(requestStorageUrl("/" + requestId));
+
+    assertThat(updatedRequest.getString("awaitingPickupRequestClosedDate"), is(notNullValue()));
```

Comment:
There are other tests that check whether a date property is similar date and time to now, using a custom matcher. 

This would provide more feedback than the property just being present.

See [this example](https://github.com/folio-org/mod-circulation/blob/666d15a77614d55ae66cde6ac07f050e3fba876f/src/test/java/api/loans/CheckInByBarcodeTests.java#L87) from mod-circulation

Reply:
Thank you for the example.
Reworked the test with a custom matcher.